### ISSUE TITLE
Binary operator '+=' cannot be applied to two 'CMTime' operands vs `shorthand_operator`

### ISSUE THREAD

**Issue by Bersaelor** [realm/SwiftLint#3643]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
error: Shorthand Operator Violation: Prefer shorthand operators (+=, -=, *=, /=) over doing the operation and assigning. (shorthand_operator)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
0.43.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
package from GitHub
* Paste your configuration file:

```swift
// This triggers a violation:
        var cumulativeTime = CMTime.zero
        while cumulativeTime + currentSegment.playbackDuration < compositionTime {
            cumulativeTime = cumulativeTime + currentSegment.playbackDuration
        }
```
Binary operator '+=' cannot be applied to two 'CMTime' operands.

Is there a way of disabling this rule for CMTime structs and similar cases of overloaded `+` ?

**Comment by SimplyDanny** [comment]
There is no way for the rule to know the types of the operands. Only the special Analyzer rules know. Thus, this is a false-positive we have to accept or disable in the code.

**Comment by SimplyDanny** [comment]
See #4953.
